src/components/TaskDetail.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import MDEditor from '@uiw/react-md-editor';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import rehypeSanitize from 'rehype-sanitize';
+
+export function TaskDetail({ task, isOpen, onClose, onUpdate, onDelete }) {
+  const { user } = useAuth();
+  const { isDark } = useTheme();
+  const [description, setDescription] = useState(task?.description || '');
+  const [isEditing, setIsEditing] = useState(false);
+  const [showDeleteConfirm, setShowDeleteConfirm] = useState(false);
+
+  useEffect(() => {
+    if (task) {
+      setDescription(task.description || '');
+    }
+  }, [task]);
+
+  if (!isOpen || !task) return null;
+
+  const handleSave = () => {
+    onUpdate(task.id, { description });
+    setIsEditing(false);
+  };
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
const handleSave = async () => {
    try {
      await onUpdate(task.id, { description });
      setIsEditing(false);
    } catch (error) {
      console.error('Failed to update task:', error);
      // Handle error appropriately (e.g., show error message)
    }
  };
```
Add error handling for onUpdate and onDelete callbacks to gracefully handle potential failures. These operations could fail due to network issues or server errors.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;